problem-solver/cxx/format-translators-module/agent/visualisation-translator-agent/scg_visualisation_translator_agent.cpp

```diff
@@ -0,0 +1,531 @@
+#include "scg_visualisation_translator_agent.hpp"
+
+#include "keynodes/format_translators_keynodes.hpp"
+
+#include <sc-agents-common/utils/IteratorUtils.hpp>
+
+#include "model/node.hpp"
+#include "model/link.hpp"
+#include "model/connector.hpp"
+
+#include "visitor/gwf_translator_visitor.hpp"
+
+namespace formatTranslators
+{
+std::string const SCgVisualisationTranslatorAgent::GWF_CONTENT_BEFORE_STATIC_SECTOR_CONTENT =
+    "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+    "<GWF version=\"2.0\">\n"
+    "    <staticSector>\n";
+std::string const SCgVisualisationTranslatorAgent::GWF_CONTENT_AFTER_STATIC_SECTOR_CONTENT =
+    "    </staticSector>\n"
+    "</GWF>\n";
+float const SCgVisualisationTranslatorAgent::CONNECTOR_IDENTIFIER_SCALE = 0.4f;
+float const SCgVisualisationTranslatorAgent::CONNECTOR_INCIDENT_POINT_PERCENT = 0.7f;
+float const SCgVisualisationTranslatorAgent::Y_INCREMENT = 100;
+float const SCgVisualisationTranslatorAgent::MIN_X = 200;
+float const SCgVisualisationTranslatorAgent::MAX_Y = 800;
+float const SCgVisualisationTranslatorAgent::MAX_LINK_WIDTH = 60;
+
+ScAddr SCgVisualisationTranslatorAgent::GetActionClass() const noexcept(false)
+{
+  return FormatTranslatorsKeynodes::action_visualise_to_scg;
+}
+
+bool SCgVisualisationTranslatorAgent::CheckInitiationCondition(ScActionInitiatedEvent const & event)
+{
+  return m_context.CheckConnector(GetActionClass(), event.GetOtherElement(), ScType::ConstPermPosArc);
+}
+
+ScResult SCgVisualisationTranslatorAgent::DoProgram(ScAction & action)
+{
+  try
+  {
+    structureToTranslate = action.GetArgument(1);
+    identifiersLanguage = action.GetArgument(2, FormatTranslatorsKeynodes::lang_en);
```

Comment:
Why is English default?

Reply:
it just is default, I am not sure that there is a reason
some language has to be default and I chose english

Reply:
Need to discuss